d3a_interface/sim_results/kpi.py

```diff
@@ -18,7 +18,7 @@
 from typing import Dict
 from d3a_interface.utils import if_not_in_list_append
 from d3a_interface.sim_results import is_load_node_type, is_buffer_node_type, \
-    is_prosumer_node_type, is_producer_node_type
+    is_prosumer_node_type, is_producer_node_type, has_no_grand_children
```

Comment:
Nit: please optimize the imports and use brackets over `\` for line continuation.